openapi-cli/src/main/java/io/ballerina/openapi/cmd/BallerinaCodeGenerator.java

```diff
@@ -339,6 +357,16 @@ private List<GenSrcFile> generateClientFiles(Path openAPI, Filter filter, boolea
         // Normalize OpenAPI definition
         OpenAPI openAPIDef = GeneratorUtils.normalizeOpenAPI(openAPI, !isResource);
         checkOpenAPIVersion(openAPIDef);
+        // Validate the service generation
+        List<String> complexPaths = GeneratorUtils.getComplexPaths(openAPIDef);
+        if (!complexPaths.isEmpty()) {
+            outStream.println("remote function(s) will be generated for client due to the given openapi definition" +
+                    " contains following complex path(s):");
+            for (String path: complexPaths) {
+                outStream.println(path);
```

Comment:
Is this a warning? (It is not started as `[warning]`)
Also are we printing the complex paths in new lines? When we print a diagnostic, the usual practice is to print a single line. Normally the stack trace is printed in multi line

Reply:
Yeah, we can include a warning here will update it. New line is added to improve content clarity for the message and we do the same while representing openapi definition parser issue.